Managers/BookmarkManager.cs

```diff
@@ -0,0 +1,85 @@
+﻿namespace ProblematicBookmark.Managers
+{
+    using Microsoft.Extensions.DependencyInjection;
+    using ProblematicBookmark.Models;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Text.RegularExpressions;
+    using System.Threading.Tasks;
+
+    internal class BookmarkManager : List<FilterResult>
+    {
+        public async Task<List<FilterResult>> Scan()
+        {
+            string informationalString = $"[info] Scan {CallerContext.BookmarkOptions.ContentRepoLocalPath}";
+            TraceLog.WriteLineInfo(informationalString);
+            //Console.WriteLine(informationalString);
+            await ScanFile(CallerContext.BookmarkOptions.ContentRepoLocalPath);
+            return this;
+        }
+
+        private async Task ScanFile(string path, int dep = 1)
+        {
+            var filePaths = Directory.GetFiles(path, "*.md");
+
+            foreach (var filePath in filePaths)
+            {
+                using (var trace = new TraceLog())
+                {
+                    try
+                    {
+                        var bookmarkurl = await GetProblematicBookmark(filePath);
+                        if (bookmarkurl != null && bookmarkurl.Count > 0)
+                        {
+                            Add(new FilterResult()
+                            {
+                                MarkdownPath = filePath,
+                                Bookmarkurl = await GetProblematicBookmark(filePath)

```

Comment:
why we need to rerun `await GetProblematicBookmark(filePath)`? can we use `bookmarkurl` directly?

Reply:
Maybe some bookmarks are right in the file. Here we should filter out the right ones